File: backend/src/main/java/com/smartcampus/backend/dto/CampusDTO.java
package com.smartcampus.backend.dto;

import lombok.*;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CampusDTO {

    private Long id;
    private String name;
    private String city;

    private Double centerLat;
    private Double centerLng;

    private Double perimeterMeters;

    // GeoJSON-style : liste de [lng, lat] pour le front (Leaflet / Mapbox)
    private List<double[]> polygonCoordinates;

    private LocalDateTime importedAt;
}

File: frontend/app/src/main/java/com/ObservatoireCampus/mobile/model/CampusDto.kt
package com.ObservatoireCampus.mobile.model


data class CampusDto(
    val id: Long,
    val name: String,
    val city: String,
    val centerLat: Double,
    val centerLng: Double,
    val perimeterMeters: Double,
    val polygonCoordinates: List<List<Double>>, // [[lng, lat], ...]
    val importedAt: String
)